projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.ts

```diff
@@ -333,9 +357,18 @@ export class IgxTextHighlightDirective implements AfterViewInit, AfterViewChecke
     public activateIfNecessary(): void {
         const group = IgxTextHighlightDirective.highlightGroupsMap.get(this.groupName);
         const column = group.columnIndex === undefined ? group.column : group.columnIndex;
-        const row = group.rowIndex === undefined ? group.row : group.rowIndex;
+        const row = group.rowIndex ? group.rowIndex : group.row;
+
+        let metadataMatch = true;
+
+        if (group.metadata) {
+            const metadata = group.metadata;
+            group.metadata.forEach((value, key) => {
+                metadataMatch = this.metadata.get(key) === value;
+            });
```

Comment:
We need to check that the map sizes are equal, also we need to break if the metadataMatch gets false. You can use the approach described here https://stackoverflow.com/questions/35948335/how-can-i-check-if-two-map-objects-are-equal.